core/src/main/c/share/vec_agg.cpp

```diff
@@ -378,9 +387,105 @@ int32_t MAX_INT(int32_t *pi, int64_t count) {
 
 #endif
 
+#ifdef SUM_SHORT
+
+int64_t SUM_SHORT(int16_t *ps, int64_t count) {
+    if (count == 0) {
+        return L_MIN;
+    }
+
+    const int32_t step = 32;
+    const auto remainder = (int32_t) (count - (count / step) * step);
+    const auto *lim = ps + count;
+    const auto *vec_lim = lim - remainder;
+
+    // instrset >=10 means AVX512BW/DQ/VL support, so it's ok to use Vec32s
+    Vec32s vec;
+    Vec16i acc0 = 0;
+    Vec16i acc1 = 0;
+    for (; ps < vec_lim; ps += step) {
+        _mm_prefetch(ps + 63 * step, _MM_HINT_T1);
+        vec.load(ps);
+        acc0 += extend_low(vec);
+        acc1 += extend_high(vec);
+    }
+
+    int64_t result = horizontal_add(acc0) + horizontal_add(acc1);
+    for (; ps < lim; ps++) {
+        int16_t v = *ps;
+        result += v;
+    }
+
+    return result;
+}
+
+int32_t MIN_SHORT(int16_t *ps, int64_t count) {
+    if (count == 0) {
+        return I_MIN;
+    }
+
+    const int step = 32;
+    const auto remainder = (int32_t) (count - (count / step) * step);
```

Comment:
nit: this looks the same as `(int32_t) (count % step)`

Reply:
Yes, but that's what we already have in _INT functions. Probably I should clean up all functions since the code looks quite inconsistent.

Reply:
Updated the expressions in 5c43c547dee665157f6409527ff8df3d7a4a8947. Decided to postpone a more significant code clean-up until the next time we touch this code.